pkg/sql/sqlinstance/instancestorage/instancestorage.go

```diff
@@ -259,6 +268,64 @@ func (s *Storage) createInstanceRow(
 				return err
 			}
 
+			regionLiveness := regionliveness.NewLivenessProber(s.db, s.codec, nil, s.settings)
+			downRegions, err := regionLiveness.QueryUnavailablePhysicalRegions(ctx, txn, false)
+			if err != nil {
+				return err
+			}
+			if downRegions.ContainsPhysicalRepresentation(string(region)) {
+				// Delete all rows for the region in system.sqlliveness, system.sql_instances,
+				// system.leases.
+				instanceSpan, err := regionliveness.GetSystemTableSpanForRegion(systemschema.SQLInstancesTable(), s.codec, string(region))
```

Comment:
nit: the implementation of the clean up should be in the region liveness package or in its own package. It doesn't really belong to sql instances or sql leases. We just need to run it before allocating a sqlliveness instance and we need sqlliveness to fail session creation if the local region is down.